TWTValidation/JSON Validator/AST Nodes/TWTJSONSchemaAmbiguousASTNode.h

```diff
@@ -0,0 +1,64 @@
+//
+//  TWTJSONSchemaAmbiguousASTNode.h
+//  TWTValidation
+//
+//  Created by Jill Cohen on 1/12/15.
+//  Copyright (c) 2015 Two Toasters, LLC.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import <TWTValidation/TWTJSONSchemaASTNode.h>
+
+
+@interface TWTJSONSchemaAmbiguousASTNode : TWTJSONSchemaASTNode
+
+@property (nonatomic, copy, readwrite) NSSet *validTypes;
+
+// Array keywords
+@property (nonatomic, strong) NSNumber *maximumItemCount;
+@property (nonatomic, strong) NSNumber *minimumItemCount;
+@property (nonatomic, assign) BOOL requiresUniqueItems;
+@property (nonatomic, assign) BOOL itemsIsSingleSchema;
```

Comment:
In the original schema, value of `items` is either a schema or an array of schemas, and an array of 1 schema has a different meaning than a single schema. (If it's an array, then during validation the first schema matches with the first item, second with the second, etc. If it's a single schema, that is used to validate all items.)

Another way to represent this would be have `node.items` be type `id` and either have an array or a node as its value, then check the class in the processor. What do you think?